properties/base/base.py

```diff
@@ -436,12 +436,51 @@ def equal(self, other):
         Equivalence is determined by checking if all Property values on
         two instances are equal, using :code:`Property.equal`.
         """
-        if self is other:
-            return True
-        if not isinstance(other, self.__class__):
-            return False
-        for prop in itervalues(self._props):
-            if prop.equal(getattr(self, prop.name), getattr(other, prop.name)):
-                continue
-            return False
+        warn('HasProperties.equal has been depricated in favor of '
+             'properties.equal and will be removed in the next release',
+             FutureWarning)
+        return equal(self, other)
+
+
+@utils.stop_recursion_with(False)
+def equal(value_a, value_b):
+    """Determine if two **HasProperties** instances are equivalent
+
+    Equivalence is determined by checking if (1) the two instances are
+    the same class and (2) all Property values on two instances are
+    equal, using :code:`Property.equal`. If the two values are the same
+    HasProperties instance (eg. :code:`value_a is value_b`) this method
+    returns True. Finally, if either value is not a HasProperties
+    instance, equality is simply checked with ==.
+    """
+    if (
+            not isinstance(value_a, HasProperties) or
+            not isinstance(value_b, HasProperties)
+    ):
+        return value_a == value_b
+    if value_a is value_b:
         return True
+    if value_a.__class__ is not value_b.__class__:
+        return False
+    for prop in itervalues(value_a._props):
```

Comment:
@lasley - I agree it would be awesome to be able to use `==`. If you dig through old commits, you'll see that is where this started... Take a look here: https://github.com/3ptscience/properties/commit/b1f2df1690abfee55bb7cc0d2b45dfd9f124f301

However, with the Python data model, there is some unwanted baggage. Overriding `__eq__` in a custom class implicitly sets `__hash__` to `None` ([see here](https://docs.python.org/3/reference/datamodel.html#object.__hash__)). This leads to unexpected behavior like this:
<img width="753" alt="screen shot 2017-03-17 at 12 19 58 pm" src="https://cloud.githubusercontent.com/assets/9453731/24057262/6a825fa4-0b0c-11e7-9dd5-224a95c55ab7.png">

My next idea was overriding `__hash__` as well but if you do it naively (ie `__hash__ = object.__hash__`), you get strange stuff like
<img width="378" alt="screen shot 2017-03-17 at 1 02 03 pm" src="https://cloud.githubusercontent.com/assets/9453731/24061051/7ceb6cb2-0b1b-11e7-97d3-1ebc31643a2a.png">

Possibly there is a way to override `__eq__` and `__hash__` that gives a more expected outcome, but this would break the suggested rule that two equal objects should hash to the same value. After spending a bit of time down this rabbit hole, I opted to take a step back and simply define a new `equal` method rather than `__eq__` (see https://github.com/3ptscience/properties/commit/b48a92715f01221ff86d360683aca7c93b4ce4dd).

I'm still open to implementing `__eq__` if there is a solution that works (I think it would at least have to break the if "`a == b` then `hash(a) == hash(b)`" rule). I'd certainly be interested to hear any additional thoughts you may have.